examples/libf3d/python/interactive-app/interactive_app.py

```diff
@@ -0,0 +1,246 @@
+import re
+from argparse import ArgumentParser
+
+import f3d
+
+
+def main(argv=None):
+    parser = ArgumentParser()
+    parser.add_argument("file")
+    parser.add_argument(
+        "--timeout",
+        type=int,
+        default=None,
+        help="Optional timeout (in seconds) before closing the viewer.",
+    )
+
+    args = parser.parse_args(argv)
+
+    f3d.Engine.autoload_plugins()
+
+    # Create a native-window engine
+    eng = f3d.Engine.create(False)
+
+    # Modify options using the struct-like API
+    opt = eng.options
+    opt.update(
+        {
+            "render.grid.enable": True,
+            "render.show_edges": True,
+            # UI overlays: axis + some HUD
+            "ui.axis": True,
+            "ui.fps": True,
+            "ui.animation_progress": True,
+            "ui.filename": True,
+            # FXAA + tone mapping
+            "render.effect.antialiasing.enable": True,
+            "render.effect.antialiasing.mode": "fxaa",
+            "render.effect.tone_mapping": True,
+        }
+    )
+
+    AddCustomCommands(eng)
+    AddCustomBindings(eng)
+
+    # Add a model to the scene
+    try:
+        eng.scene.add(args.file)
+    except Exception as e:
+        print(e)
+        return 1
+
+    # Initial render
+    win = eng.window
+    win.render()
+
+    if args.timeout:
+        # For testing purposes only, exit after `timeout` seconds
+        eng.interactor.start(args.timeout, eng.interactor.stop)
+    else:
+        eng.interactor.start()
+
+
+def parse_size(size: str):
+    if m := re.fullmatch(r"(\d+)([,x](\d+))?", size):
+        w = int(m.group(1))
+        h = int(m.group(3)) if m.group(3) else w
+        return w, h
+    else:
+        raise ValueError(f"cannot parse size from {size!r}")
+
+
+def AddCustomCommands(eng: f3d.Engine):
+    # Get the interactor
+    inter = eng.interactor
+    inter.init_commands()
+
+    # Reset options to initial state
+    opt = eng.options
+
+    # Keep a copy of the initial options
+    initial_options = dict(opt)
+
+    def reset_options(_):
```

Comment:
```suggestion
    def reset_options(_args: list[str]):
```
same for other command functions